invenio_rdm_records/services/pids/providers/base.py

```diff
@@ -47,16 +48,40 @@ def is_managed(self):
         """
         return self.system_managed
 
-    def get(self, pid_value, pid_type=None, **kwargs):
+    def get(self, pid_value, pid_type=None):
         """Get a persistent identifier for this provider.
 
-        :param pid_type: Persistent identifier type.)
+        :param pid_type: Persistent identifier type.
         :param pid_value: Persistent identifier value.
         :returns: A :class:`invenio_pidstore.models.base.PersistentIdentifier`
             instance.
         """
         return PersistentIdentifier.get(pid_type or self.pid_type, pid_value,
-                                        pid_provider=self.name, **kwargs)
+                                        pid_provider=self.name)
+
+    def get_by_record(self, record_id, pid_type=None, pid_value=None,
+                      object_type="rec"):
+        """Get a persistent identifier for this provider.
+
+        :param record_id: the record UUID.
+        :param pid_type: Persistent identifier type.
+        :param pid_value: Persistent identifier value.
+        :param object_type: Persistent identifier object_type.
+        :returns: A :class:`invenio_pidstore.models.base.PersistentIdentifier`
+            instance.
+        """
+        query = PersistentIdentifier.query.filter_by(
+            pid_type=pid_type or self.pid_type,
+            object_type=object_type,
+            object_uuid=record_id
+        )
+        if pid_value:
+            query.filter_by(pid_value=pid_value)
+
+        try:
+            return query.one()
```

Comment:
doubt: This will return the first, right? What happens in the case of "DOI/ConceptDOI"...

The record_id will be the same? (Or not because concept will be linked to the parent), then both will be `pid_type=doi` and `object_type=rec`... IIRC won't happen if `object_uuid=parent_id` for the ConceptDOI... but just to make sure.

Reply:
Yes it should never happen because of the `record_id`. I added this check to make sure that we discard the right pid for the right record, because before it was getting only by pid_type/pid_value and we might have multiple, e.g. concept.